samples/boards/nrf52/mesh/onoff-app/src/main.c

```diff
@@ -0,0 +1,694 @@
+
+	printk("Node 0x%04x received OnOff status from 0x%04x with state 0x%02x\n",
+			model->elem->addr, ctx->addr, state);
+}
+
+
+
+static int output_number(bt_mesh_output_action_t action, uint32_t number)
+{
+	printk("OOB Number: %u\n", number);
+
+	return 0;
+}
+
+static int output_string(const char *str)
+{
+	printk("OOB String: %s\n", str);
+	return 0;
+}
+
+static void prov_complete(u16_t net_idx, u16_t addr)
+{
+	printk("provisioning complete for net_idx: 0x%04x addr: 0x%04x\n",
+			net_idx, addr);
+	primary_addr = addr;
+	primary_net_idx = net_idx;
+}
+
+static void prov_reset(void)
+{
+	bt_mesh_prov_enable(BT_MESH_PROV_ADV | BT_MESH_PROV_GATT);
+}
+
+static u8_t dev_uuid[16] = { 0xdd, 0xdd };
+
+/*
+ * Button Pressed Interrupt Handler
+ */
+
+#define BUTTON_DEBOUNCE_DELAY_MS 250
+
+static void button_pressed(struct device *dev, struct gpio_callback *cb,
+		 uint32_t pins)
+{
+	time = k_uptime_get_32();
+
+	/* debounce the switch */
+	if (time < last_time + BUTTON_DEBOUNCE_DELAY_MS) {
+		return;
+	}
+
+	last_time = time;
+	sw.pin = pins;
+	k_work_submit(&sw.button_work);
+}
+
+/*
+ * Map GPIO pins to button number
+ */
+
+static uint8_t pin_to_sw(uint32_t pin)
+{
+
+	switch (pin) {
+	case BIT(SW0_GPIO_PIN): return 0;
+	case BIT(SW1_GPIO_PIN): return 1;
+	case BIT(SW2_GPIO_PIN): return 2;
+	case BIT(SW3_GPIO_PIN): return 3;
+	}
+	printk("no match for pin\n");
+	return 0;
+}
+
+/*
+ * Button Pressed Worker Task
+ */
+
+static void button_pressed_worker(struct k_work *work)
+{
+	struct bt_mesh_model *mod_cli, *mod_srv;
+	struct bt_mesh_model_pub *pub_cli, *pub_srv;
+	struct sw *sw = CONTAINER_OF(work, struct sw, button_work);
+	int err;
+	u8_t sw_idx = pin_to_sw(sw->pin);
+
+	mod_cli = mod_cli_sw[sw_idx];
+	pub_cli = mod_cli->pub;
+
+	mod_srv = mod_srv_sw[sw_idx];
+	pub_srv = mod_srv->pub;
+
+	/* Invert onoff state */
+	onoff_state[sw_idx].current = onoff_state[sw_idx].current ? 0 : 1;
+
+	printk("%s pin: 0x%08x sw: 0x%02x onoff: 0x%02x\n",
+			__func__, sw->pin, sw_idx, onoff_state[sw_idx].current);
+
+	/* pin = 0 turns on LED's on nrf52840-pca10056 board */
+	gpio_pin_write(onoff_state[sw_idx].led_device, onoff_state[sw_idx].led_gpio_pin,
+			onoff_state[sw_idx].current ? 0 : 1);
+
+	if (pub_cli->addr != BT_MESH_ADDR_UNASSIGNED) {
+		printk("%s publish to 0x%04x onoff: 0x%04x sw_idx: 0x%04x\n",
+				__func__, pub_cli->addr,
+				onoff_state[sw_idx].current, sw_idx);
+		bt_mesh_model_msg_init(pub_cli->msg,
+				BT_MESH_MODEL_OP_GEN_ONOFF_SET);
+		net_buf_simple_add_u8(pub_cli->msg,
+				onoff_state[sw_idx].current);
+		net_buf_simple_add_u8(pub_cli->msg, trans_id++);
+		err = bt_mesh_model_publish(mod_cli);
+		if (err) {
+			printk("%s bt_mesh_model_publish: err: %d\n",
+					__func__, err);
+		}
+	}
+
+	/*
+	 * If the LED's server is bound, send status
+	 *
+	 * If model has at least one app key, it is bound
+	 *
+	 * We probably don't need to send status to ourself,
+	 * but this is just an example.
+	 *
+	 */
+#if 0
```

Comment:
Is there a good reason why you're keeping this code block around? What does the spec say about this? Either we need the code and it should be uncommented, or then it should just be removed, IMO.

Reply:
i removed it.